### ISSUE TITLE
[Bug] Newsletter Categories: Subscriptions not showing in reader for some WPCOM user

### ISSUE THREAD

**Issue by crisbusquets** [Automattic/wp-calypso#83099]
From Slack thread: p1697535841812949-slack-C02TCEHP3HA.

I (ileo.red@gmail.com) am subscribed to [songokunewsletter.wordpress.com](http://songokunewsletter.wordpress.com/).

![Image](https://github.com/Automattic/wp-calypso/assets/25607244/26ed5cab-61e6-4912-946a-b78d14e948d4)

When I log in to WordPress.com with that user, my Reader (Lector) is empty.
I wanted to manage my subscription to that blog, and I can't (or I can't find where to do it).

https://github.com/Automattic/wp-calypso/assets/25607244/dffd98d4-1156-4343-9505-39a409cef60a

**Comment by crisbusquets** [comment]
Another related issue with the same user (ileo.red@gmail.com):

* I manually subscribed to https://yetanothernewslettercris.wordpress.com/
* I confirmed my subscription
* On the confirmation email, I clicked `Manage subscriptions`, which opened /read/subscriptions for that blog

This is what I see. In short, it says that a blog called `yetanothernewslettercris.wordpress.com` doesn't exist (No se encontró ningún resultado para...), but then it suggests that same blog right below. With an option to subscribe, but I'm already subscribed.

<img width="1508" alt="Screenshot 2023-10-17 at 15 59 38" src="https://github.com/Automattic/wp-calypso/assets/25607244/d1707aea-f42c-4877-966b-6e31b5738d6f">
